### PR TITLE
CIV-4393: Briefly explain your claim

### PR DESCRIPTION
**Pull request by portizdev**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-4393

### Change description ###
- Creating a service for claims details
- Refactoring detailsOfClaim into ClaimDetails - Reason
- Screen: 'Briefly explain your claim'.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by laarmada — APPROVED** [review]

File: src/main/services/features/claim/details/claimDetailsService.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  }
+};
+
+const saveClaimDetails = async (claimId: string, value: any, claimDetailsPropertyName: string): Promise<void> => {
+  try {
+    const claim: any = await getCaseDataFromStore(claimId);
+    if (claim.claimDetails) {
+      claim.claimDetails[claimDetailsPropertyName] = value;
+    } else {
+      const claimDetails: any = new ClaimDetails();
```
Comment (GorilaNaranja) [inline src/main/services/features/claim/details/claimDetailsService.ts:24]: Why any and not ClaimDetails?

File: src/main/routes/features/claim/details/reasonController.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import {
+  CLAIM_REASON_URL,
+  CLAIM_TIMELINE_URL,
+} from '../../../urls';
+import {getClaimDetails, saveClaimDetails} from '../../../../services/features/claim/details/claimDetailsService';
+import {Reason} from '../../../../common/form/models/claim/details/reason';
+
+const reasonController = Router();
+const reasonViewPath = 'features/claim/details/reason';
+const dqPropertyName = 'reason';
```
Comment (laperla11) [inline src/main/routes/features/claim/details/reasonController.ts:14]: can you change it to another name like claimDetailsProprtyName?

File: src/main/services/features/claim/details/claimDetailsService.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {getCaseDataFromStore, saveDraftClaim} from '../../../../modules/draft-store/draftStoreService';
+import {ClaimDetails} from '../../../../common/form/models/claim/details/claimDetails';
+
+const {Logger} = require('@hmcts/nodejs-logging');
+const logger = Logger.getLogger('claimDetailsService');
+
+const getClaimDetails = async (claimId: string): Promise<ClaimDetails> => {
+  try {
+    const caseData = await getCaseDataFromStore(claimId);
+    return Object.assign(new ClaimDetails(), caseData.claimDetails);
```
Comment (laperla11) [inline src/main/services/features/claim/details/claimDetailsService.ts:10]: Is it not possible to return the caseData.claimDetails here without using Object.assign, becasue you're doing another Object.assign in the controller after this step.

**Review by laperla11 — APPROVED** [review]

File: src/main/services/features/claim/details/claimDetailsService.ts (hunk 3 above)
Comment (portizdev) [reply src/main/services/features/claim/details/claimDetailsService.ts:11]: True! Done!

File: src/main/routes/features/claim/details/reasonController.ts (hunk 2 above)
Comment (portizdev) [reply src/main/routes/features/claim/details/reasonController.ts]: Done! Consequences of copy-paste

**Review by GorilaNaranja — APPROVED** [review]

File: src/main/services/features/claim/details/claimDetailsService.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
+};
+
+const saveClaimDetails = async (claimId: string, value: any, claimDetailsPropertyName: string): Promise<void> => {
```
Comment (GorilaNaranja) [inline src/main/services/features/claim/details/claimDetailsService.ts:18]: Good job removing "any", maybe you can use same approach for value:

`value: keyof ClaimDetails`

**Pull request merged**